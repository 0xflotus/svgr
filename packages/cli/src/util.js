/* eslint-disable no-console */
import fs from 'fs'
import chalk from 'chalk'
import util from 'util'
import svgrConvert from '@svgr/core'
import svgo from '@svgr/plugin-svgo'
import jsx from '@svgr/plugin-jsx'
import prettier from '@svgr/plugin-prettier'

export const readFile = util.promisify(fs.readFile)
export const stat = util.promisify(fs.stat)

export function convert(code, config, state) {
  return svgrConvert.sync(code, config, {
    ...state,
    caller: {
      name: '@svgr/cli',
      defaultPlugins: [svgo, jsx, prettier],
    },
  })
}

export async function convertFile(filePath, config = {}) {
  const code = await readFile(filePath, 'utf-8')
  return convert(code, config, { filePath })
}

export function exitError(error) {
  console.error(chalk.red(error))
  process.exit(1)
}
