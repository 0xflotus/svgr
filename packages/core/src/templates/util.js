export const getProps = config => {
  const props = []
  if (config.ref) props.push('svgRef')
  if (config.titleProp) props.push('title')
  if (config.expandProps) props.push('...props')

  if (props.length === 0) return '()'
  if (props.length === 1 && config.expandProps) return 'props'

  return `({ ${props.join(', ')} })`
}

export const getExport = (config, state) => {
  const component = config.ref ? 'ForwardRef' : state.componentName
  if (state.webpack && state.webpack.previousExport) {
    let result = ''
    result += `export default ${state.webpack.previousExport}\n`
    result += `export { ${component} as ReactComponent }`
    return result
  }
  if (state.rollup && state.rollup.previousExport) {
    let result = ''
    result += `${state.rollup.previousExport}\n`
    result += `export { ${component} as ReactComponent }`
    return result
  }
  return `export default ${component}`
}

export const getForwardRef = (config, state) => {
  if (!config.ref) return ''
  return `const ForwardRef = React.forwardRef((props, ref) => <${
    state.componentName
  } svgRef={ref} {...props} />)\n\n`
}
