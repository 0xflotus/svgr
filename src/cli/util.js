/* eslint-disable no-console */
import chalk from 'chalk'
import { extname } from 'path'
import fs from 'mz/fs'
import readdir from 'recursive-readdir'
import { rawConvert } from '../'

export { rawConvert }

export async function convertFile(filePath, opts) {
  const code = await fs.readFile(filePath, 'utf-8')
  return rawConvert(code, opts, { filePath })
}

export function exitError(error) {
  console.error(chalk.red(error))
  process.exit(1)
}

export { readdir }

export async function readdirFilter(dirname) {
  return (await readdir(dirname)).filter(isCompilableExtension)
}

const COMPILABLE_EXTENSIONS = ['.svg', '.SVG']

export function isCompilableExtension(filename) {
  const ext = extname(filename)
  return COMPILABLE_EXTENSIONS.includes(ext)
}
