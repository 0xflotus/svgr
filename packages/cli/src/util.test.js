import path from 'path'
import { convertFile } from './util'

const FIXTURES = path.join(__dirname, '../../../__fixtures__')

describe('util', () => {
  describe('#convertFile', () => {
    it('should convert a file', async () => {
      const file = path.join(FIXTURES, 'simple/file.svg')
      const result = await convertFile(file)
      expect(result).toMatchSnapshot()
    })

    it('should support a custom config path', async () => {
      const file = path.join(FIXTURES, 'simple/file.svg')
      const result = await convertFile(file, {
        config: '__fixtures__/withSvgrRc/.svgrrc',
      })
      expect(result).toMatchSnapshot()
    })
  })
})
