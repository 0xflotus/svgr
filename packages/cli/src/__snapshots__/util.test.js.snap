// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util #convertFile should convert a file 1`] = `
"import React from 'react'

const File = props => (
  <svg width={48} height={1} {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default File
"
`;

exports[`util #convertFile should support a custom config path 1`] = `
"import React from 'react'

const File = props => (
  <svg viewBox=\\"0 0 48 1\\" width=\\"1em\\" height=\\"1em\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default File
"
`;
